From: Steven Rostedt <srostedt@redhat.com>
Subject: Change bug_on for atomic to pagefault_disabled.

The lockless page changes decoupled the pagefault disabled
from preempt count. The bug_on in filemap.c is now not correct.
This patch uses the proper check.

Signed-off-by: Steven Rostedt <srostedt@redhat.com>
---
 mm/filemap.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: linux-2.6.26/mm/filemap.c
===================================================================
--- linux-2.6.26.orig/mm/filemap.c
+++ linux-2.6.26/mm/filemap.c
@@ -1804,7 +1804,7 @@ size_t iov_iter_copy_from_user_atomic(st
 	size_t copied;
 
 #ifndef CONFIG_PREEMPT_RT
-	BUG_ON(!in_atomic());
+	BUG_ON(!current->pagefault_disabled);
 #endif
 	kaddr = kmap_atomic(page, KM_USER0);
 	if (likely(i->nr_segs == 1)) {
